internal/conversion/validate_vni.go

```diff
@@ -33,16 +33,33 @@ func ValidateL2VNIs(l2Vnis []v1alpha1.L2VNI) error {
 		return err
 	}
 
-	// Perform L2-specific validation (HostMaster and L2GatewayIP validation)
+	// Perform L2-specific validation (HostMaster and L2GatewayIPs validation)
 	for _, vni := range l2Vnis {
 		if vni.Spec.HostMaster != nil && vni.Spec.HostMaster.Name != "" {
 			if err := isValidInterfaceName(vni.Spec.HostMaster.Name); err != nil {
 				return fmt.Errorf("invalid hostmaster name for vni %s: %s - %w", vni.Name, vni.Spec.HostMaster.Name, err)
 			}
 		}
-		if vni.Spec.L2GatewayIP != "" {
-			if err := isValidCIDR(vni.Spec.L2GatewayIP); err != nil {
-				return fmt.Errorf("invalid l2gatewayip for vni %s: %s - %w", vni.Name, vni.Spec.L2GatewayIP, err)
+		if len(vni.Spec.L2GatewayIPs) > 2 {
+			return fmt.Errorf("invalid l2gatewayips for vni %s: %v - should containt at most two CIDRs", vni.Name, vni.Spec.L2GatewayIPs)
+		}
+
+		for _, ip := range vni.Spec.L2GatewayIPs {
+			if err := isValidCIDR(ip); err != nil {
+				return fmt.Errorf("invalid l2gatewayips for vni %s: %s - %w", vni.Name, ip, err)
+			}
+		}
+		if len(vni.Spec.L2GatewayIPs) == 2 {
+			firstIP, _, err := net.ParseCIDR(vni.Spec.L2GatewayIPs[0])
+			if err != nil {
+				return err
+			}
+			secondIP, _, err := net.ParseCIDR(vni.Spec.L2GatewayIPs[1])
+			if err != nil {
+				return err
+			}
+			if (firstIP.To4() != nil) == (secondIP.To4() != nil) {
```

Comment:
use the ipfamily package here

Reply:
> use the ipfamily package here

I will refactor ipfamily so we can use the multiple addresses check with CIDRs